### PR TITLE
Use prior to get ensemble size

### PR DESCRIPTION
**Pull request by oyvindeide**
**Issue**
Resolves #my_issue

**Approach**
_Short description of the approach_

(Screenshot of new behavior in GUI if applicable)

## Pre review checklist

- [ ] Read through the code changes carefully after finishing work
- [ ] Make sure tests pass locally (after every commit!)
- [ ] Prepare changes in small commits for more convenient review (optional)
- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Updated documentation
- [ ] Ensured that unit tests are added for all new behavior (See 
    [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)),
    and changes to existing code have good test coverage.

## Pre merge checklist
- [ ] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: src/ert/run_models/iterated_ensemble_smoother.py [hunk 1]
```diff
@@ -130,7 +130,7 @@ def run_experiment(
             posterior = self._storage.create_ensemble(
                 self._experiment_id,
                 name=target_case_format % current_iter,  # noqa
-                ensemble_size=self.ert.getEnsembleSize(),
+                ensemble_size=prior_context.sim_fs.ensemble_size,
```
Comment (kvashchuka) [inline src/ert/run_models/iterated_ensemble_smoother.py:133]: Why do we want to change it? For `prior` we still will have to use `ensemble_size=self.ert.getEnsembleSize()`, or could ensemble size change from prior to posterior if some of the realizations failed?

Comment (oyvindeide) [reply src/ert/run_models/iterated_ensemble_smoother.py:133]: For prior it is fine, as we then start from scratch, however for the posterior we start from the prior, and not the configuration. The size does not change even if realizations fail, but think it is better not to rely on external state.

Comment (kvashchuka) [reply src/ert/run_models/iterated_ensemble_smoother.py:133]: Agere

**Review by kvashchuka — APPROVED** [review]
LGTM

**Pull request merged**